src/index.spec.ts

```diff
@@ -213,88 +164,4 @@ describe('EppoClient E2E test', () => {
     const assignment = client.getAssignment('subject-10', flagKey, subjectAttributes);
     expect(assignment).toEqual('control');
   });
-
-  it('only returns variation if subject matches rules', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
-    const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.configStore.get(flagKey)).thenReturn({
-      ...mockExperimentConfig,
-      rules: [
-        {
-          allocationKey: 'allocation1',
-          conditions: [
-            {
-              operator: OperatorType.GT,
-              attribute: 'appVersion',
-              value: 10,
-            },
-          ],
-        },
-      ],
-    });
-    const client = new EppoClient(mockConfigRequestor, mockPoller);
-    let assignment = client.getAssignment('subject-10', flagKey, { appVersion: 9 });
-    expect(assignment).toEqual(null);
-    assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual(null);
-    assignment = client.getAssignment('subject-10', flagKey, { appVersion: 11 });
-    expect(assignment).toEqual('control');
-  });
-
-  function getAssignmentsWithSubjectAttributes(
-    subjectsWithAttributes: {
-      subjectKey: string;
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      subjectAttributes?: Record<string, any>;
-    }[],
-    experiment: string,
-    valueTestType: ValueTestType = ValueTestType.StringType,
-  ): (EppoValue | null)[] {
-    const client = getInstance();
-    return subjectsWithAttributes.map((subject) => {
-      switch (valueTestType) {
-        case ValueTestType.BoolType: {
-          const ba = client.getBoolAssignment(
```

Comment:
we're calling the top-level functions here so really we shouldn't need to know or care about `EppoValue`